fiftyone/plugins/secrets.py

```diff
@@ -128,32 +146,36 @@ def __getitem__(self, key):
                 self.__secrets[key] = val
         return val
 
+    def __setattr__(self, key, value):
+        if re.search("_SecretsDictionary__(secrets|required_keys)$", key):
+            raise KeyError("Cannot mutate hidden properties")
+        else:
+            super().__setattr__(key, value)
+
     def __setitem__(self, key, value):
         raise RuntimeError("Setting values is not allowed")
 
     def __deepcopy__(self, memodict={}):
-        logging.warning(
-            "Copying the SecretDictionary values is not " "allowed."
-        )
+        logging.warning("Copying the SecretsDictionary values is not allowed.")
```

Comment:
Why log a warning that something is now allowed but let it happen anyway?

Also why is this method implemented? It returns a regular dict so is not actually copying the object, it returns a different type of object.

Why not let the default implementations of `copy()` and `__deepcopy__` do their things?

Reply:
To discourage copying the secret values and passing them around and encourage usage exclusively via ctx.secret/s both for security reasons and to avoid potential issues with copying the value to an object and expecting that the object have the secrets when called in another part of the code due to the execution context potentially being reset inbetween